src/boosting/gbdt.cpp

```diff
@@ -107,10 +105,14 @@ void GBDT::Init(const Config* config, const Dataset* train_data, const Objective
 
   num_data_ = train_data_->num_data();
   // create buffer for gradients and Hessians
+  size_t total_size = static_cast<size_t>(num_data_) * num_tree_per_iteration_;
   if (objective_function_ != nullptr) {
-    size_t total_size = static_cast<size_t>(num_data_) * num_tree_per_iteration_;
     gradients_.resize(total_size);
     hessians_.resize(total_size);
+  } else {
+    // use customized objective function, only for GOSS
```

Comment:
Please consider moving this into the GOSS Sample strategy. For example, we can do this in `ResetConfig`.

Reply:
Resolved.